### PR TITLE
XML export and import support in DesignSpaceHelper API

### PR DESCRIPTION
**Pull request by jonathonw**

### CODE REVIEW

File: src/DesignSpaceHelper/ComponentLib.idl [hunk 1]
```diff
@@ -16,6 +16,8 @@ import "mga.idl";
 interface IDesignSpaceHelper : IMgaComponentEx
 {
 	HRESULT ApplyConstraintsAndGenerateCWCs([in] IMgaProject *project, [in] IMgaFCO *currentobj, [in] VARIANT_BOOL applyConstraints);
+	HRESULT ExportDesertXML([in] IMgaProject *project, [in] IMgaFCO *currentobj, [in] BSTR exportPath);
```
[the remark this answers is not in the document]
Comment (jonathonw) [reply src/DesignSpaceHelper/ComponentLib.idl]: Is it correct that the interface UUID should change?  [This](https://docs.microsoft.com/en-us/windows/desktop/midl/version) seems to imply that, when adding new functions to the interface, the interface UUID should remain the same and the minor version should be the same.

[the remark this answers is not in the document]
Comment (ksmyth) [reply src/DesignSpaceHelper/ComponentLib.idl]: That link isn't talking about COM objects.

It says: "an interface attribute list that includes the [object] attribute cannot include the [version] attribute"

https://docs.microsoft.com/en-us/windows/desktop/midl/object says "The [object] interface attribute identifies a COM interface. (An interface attribute list that does not include the [object] attribute indicates a DCE RPC interface.)"

[the remark this answers is not in the document]
Comment (jonathonw) [reply src/DesignSpaceHelper/ComponentLib.idl]: Makes sense.

Should we bump the major version or minor version here?

[the remark this answers is not in the document]
Comment (ksmyth) [reply src/DesignSpaceHelper/ComponentLib.idl]: I don't think it really matters. I'd prefer minor version

**Pull request merged**